ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetDetailedObjects.java

```diff
@@ -62,55 +68,120 @@ public CliCommand init(final Arguments args) throws Exception {
-            return null;
+            return Predicates.notNull();
         }
 
-        // else build a predicate from search-params
-        final Map<String, String> ranges = parseMeta();
-        final long largerthan = Long.parseLong(ranges.get(LARGERTHAN));
-        final long smallerthan = Long.parseLong(ranges.get(SMALLERTHAN));
-        final Date newerthan = new Date(Long.parseLong(ranges.get(NEWERTHAN)));
-        final Date olderthan = new Date(Long.parseLong(ranges.get(OLDERTHAN)));
+        String newer = this.filterParams.get(NEWERTHAN);
+        String older = this.filterParams.get(OLDERTHAN);
+        if (Guard.isStringNullOrEmpty(newer) && Guard.isStringNullOrEmpty(older)) {
+            return Predicates.notNull();
+        }
+        // if one is specified, default the other
+        final long newerThan = Guard.isStringNullOrEmpty(newer) ? 0L : new Date().getTime() - Utils.dateDiffToSeconds(newer) * 1000;
+        final long olderThan = Guard.isStringNullOrEmpty(older) ? Long.MAX_VALUE : new Date().getTime() - Utils.dateDiffToSeconds(older) * 1000;
 
         return new Predicate<DetailedS3Object>() {
             @Override
             public boolean apply(@Nullable final DetailedS3Object input) {
-                return input.getSize() > largerthan
-                        && input.getSize() < smallerthan
-                        && input.getCreationDate().after(newerthan)
-                        && input.getCreationDate().before(olderthan);
+                return input.getCreationDate().after(new Date(newerThan))
+                        && input.getCreationDate().before(new Date(olderThan));
+            }
+        };
+    }
+
+    protected Predicate<DetailedS3Object> getOwnerPredicate() throws CommandException {
+        if (Guard.isMapNullOrEmpty(this.filterParams)) {
+            return Predicates.notNull();
+        }
+        final String owner = metaLookup(OWNER);
+        if (Guard.isStringNullOrEmpty(owner)) {
+            return Predicates.notNull();
+        }
+        return new Predicate<DetailedS3Object>() {
+            @Override
+            public boolean apply(@Nullable final DetailedS3Object input) {
+                return input.getOwner().equals(owner);
             }
         };
     }
 
+    protected Predicate<DetailedS3Object> getNamePredicate() throws CommandException {
+        if (Guard.isMapNullOrEmpty(this.filterParams)) {
+            return Predicates.notNull();
+        }
+        final String contains = metaLookup(CONTAINS);
+        if (Guard.isStringNullOrEmpty(contains)) {
+            return Predicates.notNull();
+        }
+        return new Predicate<DetailedS3Object>() {
+            @Override
+            public boolean apply(@Nullable final DetailedS3Object input) {
+                return input.getName().contains(contains);
+            }
+        };
+    }
+
+    private String metaLookup(final String key) throws CommandException {
```

Comment:
What exactly is meant by `meta` in this method name?  This seems like it could be named better, or needs some comments to explain what's meta.